build-system/tasks/e2e/build-bento-e2e.js

```diff
@@ -0,0 +1,132 @@
+ *
+ * @param {!Object} options
+ * @return {!Promise<void>}
+ */
+async function buildBentoE2E(options) {
+  const e2eFixtures = await fastGlob(
+    'extensions/*/1.0/test-e2e/e2e-fixture.js'
+  );
+
+  const isMissingCss =
+    (await fastGlob(`src/bento/components/*/1.0/dist/styles.css`)).length === 0;
+  if (isMissingCss) {
+    await compileCss(options);
+    await buildExtensions({...options, compileOnlyCss: true});
+  }
+
+  const results = e2eFixtures.flatMap((fixturePath) => {
+    const newOptions = {
+      ...options,
+      // outputting builds to test directory
+      outputPath: 'test/fixtures/e2e/',
+    };
+    const bentoComponentName = fixturePath.match(/amp-(.*(?=\/\d))/)?.[1];
+
+    if (!bentoComponentName) {
+      throw new Error('component name not found for e2e testing');
+    }
+    const buildCssForTests = async () => {
+      const bentoPath = `src/bento/components/bento-${bentoComponentName}/1.0`;
+      const distPath = `${bentoPath}/dist`;
+      if (!existsSync(distPath)) {
+        mkdirSync(distPath);
+      }
+      if (!existsSync(`${distPath}/styles.css`)) {
+        return buildNpmCss(bentoPath, {
+          ...options,
+          name: `bento-${bentoComponentName}`,
+          version: '1.0',
+          bento: true,
+        });
+      }
+    };
+
+    return [
+      // build the js bundles to test
+      buildBinaries(
+        `extensions/amp-${bentoComponentName}/1.0/test-e2e`,
+        [
+          // Build preact options
+          {
+            entryPoint: 'e2e-fixture.js',
+            outfile: `bento-${bentoComponentName}-e2e-build-preact.js`,
+            wrapper: '',
+          },
+          // Build react options
+          {
+            babelCaller: 'react-unminified',
+            entryPoint: 'e2e-fixture.js',
+            external: [],
+            outfile: `bento-${bentoComponentName}-e2e-build-react.js`,
+            wrapper: '',
+            remap: {
+              'preact': 'react',
+              '.*/preact/compat': 'react',
+              'preact/hooks': 'react',
+              'preact/dom': 'react-dom',
+            },
+          },
+        ],
+        newOptions
+      ),
+      buildCssForTests(),
+      // build the html pages to load the e2e SPA
+      buildE2eTestPage(bentoComponentName, 'preact', newOptions),
+      buildE2eTestPage(bentoComponentName, 'react', newOptions),
+    ];
+  });
+  await Promise.all(results);
+}
+
+/**
+ * Creates html for an SPA to end-to-end test bento component with preact or react.
+ *
+ * @param {string} bentoComponentName
+ * @param {'preact' | 'react'} buildFor
+ * @param {Object} options
+ * @return {Promise<void>}
+ */
+async function buildE2eTestPage(bentoComponentName, buildFor, options) {
+  const startTime = Date.now();
+  const htmlTemplate = `<!DOCTYPE html>
+    <html>
+      <head>
+        <script async src="./bento-${bentoComponentName}-e2e-build-${buildFor}${
+    options.minify ? '' : '.max'
+  }.js"></script>
```

Comment:
nit cause I'm not a big fan of line-breaking a formatting string:
```suggestion
  const buildForExt = options.minify ? buildFor : `${buildFor}.max`;
  const htmlTemplate = `<!DOCTYPE html>
    <html>
      <head>
        <script async src="./bento-${bentoComponentName}-e2e-build-${buildForExt}.js"></script>